src/addons/Pagination/PaginationItem.js

```diff
@@ -64,19 +67,26 @@ class PaginationItem extends Component {
     if (keyboardKey.getCode(e) === keyboardKey.Enter) _.invoke(this.props, 'onClick', e, this.props)
   }
 
+  handleOverrides = () => ({
+    onClick: this.handleClick,
+    onKeyDown: this.handleKeyDown,
+  })
+
   render() {
-    const { active, ariaLabel, type, ...rest } = this.props
-    const disabled = type === 'ellipsisItem'
-
-    return MenuItem.create({
-      ...rest,
-      active,
-      'aria-current': active,
-      'aria-label': ariaLabel,
-      disabled,
-      onClick: this.handleClick,
-      onKeyDown: this.handleKeyDown,
-      tabIndex: disabled ? -1 : 0,
+    const { active, ariaLabel, type } = this.props
+    const disabled = this.props.disabled || type === 'ellipsisItem'
+
+    return MenuItem.create(this.props, {
+      defaultProps: {
+        active,
+        disabled,
+        'aria-current': active,
+        'aria-label': ariaLabel,
+        onClick: this.handleClick,
+        onKeyDown: this.handleKeyDown,
+        tabIndex: disabled ? -1 : 0,
+      },
+      overrideProps: this.handleOverrides,
```

Comment:
@levithomason I've pushed some changes and added some new tests, ready for your review :+1:

Reply:
You're awesome ❤️